routes/api_v2_0.py

```diff
@@ -441,6 +447,126 @@ def class4_data(
+                        dims=db.get_all_dimensions()
+        else:
+                if not isinstance(config.url, list):
+                        data = xr.open_mfdataset([config.url])
+                else:
+                        data = xr.open_mfdataset(config.url)
+                dims=data.dims
+        has_depth="depth" in dims
+        if config.model_class != "Nemo":
+                vector_variables = list(config.vector_variables.keys())
+        else:
+            vector_variables=[]
+
+        for variable in config.variables:
+            variable_name=config.variable[variable].name
+            scale = config.variable[variable].scale
+
+            entry={
+                "dataset_id":dataset_key,
+                "variable_id":variable,
+                "variable_scale":scale,
+                "vector_variables":vector_variables,
+                "depth":has_depth
+            }
+
+            if variable_name in variables:
+                variables[variable_name].append(entry)
+            else:
+                variables[variable_name]=[entry]
+    return variables
+
+@router.get("/datasets/filter/date")
+def filter_datasets_by_date(
+    target_date: str = Query(description="Target date in ISO format"),
+    dataset_ids: str = Query(description="Comma-separated dataset IDs to filter from"),
+):
+    """
+    Returns only matching dataset IDs for date filter.
+    """
+    dataset_id_list = [id.strip() for id in dataset_ids.split(",") if id.strip()]
+
+    matching_dataset_ids = []
+    parsed_date = dateparse(target_date)
+
+    for dataset_id in dataset_id_list:
+        config = DatasetConfig(dataset_id)
+        if not isinstance(config.url, list) and config.url.endswith(".sqlite3"):
+            with SQLiteDatabase(config.url) as db:
+                vals = np.array(db.get_all_timestamps())
+                time_dim_units = config.time_dim_units
+                lowest=np.min(vals)
+                highest=np.max(vals)
+                converted_times = time_index_to_datetime([lowest,highest], time_dim_units)
+                if (
+                    converted_times[0] <= parsed_date
+                    and converted_times[1] >= parsed_date
+                ):
+                    matching_dataset_ids.append(dataset_id)
+        else:
+            if not isinstance(config.url, list):
+                data = xr.open_mfdataset([config.url])
+            else:
+                data = xr.open_mfdataset(config.url)
+            time_vals = data.time.values
+            first_timestamp = time_vals[0].astype("datetime64[s]").astype(int)
+            last_timestamp = time_vals[-1].astype("datetime64[s]").astype(int)
+            first_time = datetime.fromtimestamp(first_timestamp, tz=timezone.utc)
+            last_time = datetime.fromtimestamp(last_timestamp, tz=timezone.utc)
+            if first_time <= parsed_date and last_time >= parsed_date:
+                matching_dataset_ids.append(dataset_id)
```

Comment:
Rather than going through the trouble of converting the timestamps to a datetime object you can just do the inverse:
```suggestion
            if time_vals.min() <= np.datetime64(parsed_date) and time_vals.max() >= np.datetime64(parsed_date):
                matching_dataset_ids.append(dataset_id)
```